code/game/gamemodes/changeling/powers/transform.dm

```diff
@@ -11,36 +11,93 @@
 	name = "flesh"
 	flags = NODROP
 
+/obj/item/clothing/glasses/changeling/attack_hand(mob/user)
+	if(loc == user && user.mind && user.mind.changeling)

```

Comment:
instead of doing this via qdel, set the NODROP flag to off and set DROPDEL and then just let the parent proc run and it should have the same effect without requiring a forced qdel and doing snowflake things to sidestep NODROP